sycl/include/CL/sycl/interop_handle.hpp

```diff
@@ -0,0 +1,117 @@
+//==------------ interop_handle.hpp --- SYCL interop handle ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/access/access.hpp>
+#include <CL/sycl/detail/accessor_impl.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/detail/pi.hpp>
+
+#include <memory>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+namespace detail {
+class AccessorBaseHost;
+class ExecCGCommand;
+class DispatchHostTask;
+} // namespace detail
+
+template <typename DataT, int Dims, access::mode AccMode,
+          access::target AccTarget, access::placeholder isPlaceholder>
+class accessor;
+
+class interop_handle {
+public:
+  /// Receives a SYCL accessor that has been defined is a requirement for the
+  /// command group, and returns the underlying OpenCL memory object that is
+  /// used by the SYCL runtime. If the accessor passed as parameter is not part
+  /// of the command group requirements (e.g. it is an unregistered placeholder
+  /// accessor), the exception `cl::sycl::invalid_object` is thrown
+  /// asynchronously.
+  template <typename dataT, int dimensions, access::mode accessmode,
```

Comment:
Instead of making this OpenCL specific it would be better to use the existing interop to make it work for all backends. `interop_handler` uses this in the `get_mem` and `get_queue` methods.

The same goes for `get_native_queue`, `get_native_device`, and `get_native_context`.

Reply:
Notice that you may have to add a specialization of `interop` for accessors with `access::target::host_task` inside sycl/include/CL/sycl/backend/opencl.hpp.

Reply:
Will it align with the proposal? https://github.com/codeplaysoftware/standards-proposals/blob/master/host_task/host_task.md

Reply:
I believe there is a section about it at the end of the proposal, but even if there wasn't we could default the backend to OpenCL, so the user hardly wouldn't know the difference.

Reply:
On the other hand, should it return `pi_*` types instead of backend-specific?

Reply:
From my understanding, users should not be exposed to PI objects.

Reply:
OK, got it.

Reply:
I've untied `interop_handle` methods from OpenCL types.

Reply:
Looks good!